src/components/flowchart/components/sliced-pipeline-action-bar/sliced-pipeline-action-bar.js

```diff
@@ -1,27 +1,102 @@
+ */
+const calculateTransformX = ({
+  screenWidth,
+  slicePipelineActionBarWidth,
+  metaDataPanelWidth,
+  nodeListWidth,
+  minimumTransformX,
 }) => {
-  const { outerWidth } = chartSize;
+  const actionBarWidthAdjustment =
+    screenWidth > 2200
+      ? slicePipelineActionBarWidth
+      : slicePipelineActionBarWidth / 2;
+  return Math.max(
+    screenWidth - nodeListWidth - metaDataPanelWidth - actionBarWidthAdjustment,
+    minimumTransformX
+  );
+};
+
+export const SlicedPipelineActionBar = React.forwardRef((props, ref) => {
+  const {
+    chartSize,
+    displayMetadataPanel,
+    isSlicingPipelineApplied,
+    onApplySlicingPipeline,
+    onResetSlicingPipeline,
+    runCommand,
+    slicedPipeline,
+    visibleSidebar,
+  } = props;
+  const { outerWidth: screenWidth } = chartSize;
+  const buffer = 200;
+  const slicePipelineActionBarWidth =
+    ref.current && ref.current.firstChild.getBoundingClientRect().width;
+  const metaDataPanelWidth = displayMetadataPanel
+    ? metaSidebarWidth.open + buffer
+    : metaSidebarWidth.open;
+  const nodeListWidth = visibleSidebar
+    ? sidebarWidth.open - buffer
+    : sidebarWidth.open - buffer / 2;
+  const minimumTransformX = visibleSidebar
+    ? sidebarWidth.open
+    : sidebarWidth.closed;
+
+  const transformX = calculateTransformX({
+    screenWidth,
+    slicePipelineActionBarWidth,
+    metaDataPanelWidth,
+    nodeListWidth,
+    minimumTransformX,
+  });
 
-  const transformX = visibleSidebar ? outerWidth / 2 + 100 : outerWidth / 2;
   return (
     <div
       className="sliced-pipeline-action-bar"
-      style={{ transform: `translateX(${transformX}px)` }}
+      style={{ transform: `translateX(${transformX}px)`, opacity: 1 }}
     >
       <div className="sliced-pipeline-action-bar--info">
-        {`${slicedPipeline.length} selected`}
+        {`${slicedPipeline.length} ${
+          isSlicingPipelineApplied ? 'sliced' : 'selected'
+        }`}
       </div>
-      <div className="sliced-pipeline-action-bar--cta">
-        <Button>Slice</Button>
+      <div
+        className={classnames('sliced-pipeline-action-bar--run-command', {
+          'sliced-pipeline-action-bar--run-command-long':
+            runCommand.length > 85,
+        })}
+      >
+        <CommandCopier command={runCommand} isCommand={true} />
       </div>
+      {isSlicingPipelineApplied ? (
+        <div className="sliced-pipeline-action-bar--cta sliced-pipeline-action-bar--reset">
+          <Button onClick={onResetSlicingPipeline}>Reset slice</Button>
+        </div>
+      ) : (
+        <div
+          className="sliced-pipeline-action-bar--cta sliced-pipeline-action-bar--slice"
+          onClick={onApplySlicingPipeline}
+        >
+          <IconButton
+            ariaLabel="Copy run command to clipboard."
+            className="copy-button"
+            dataTest={`clicked.run_command`}
```

Comment:
need to update in contributing docs but the norm is dataTest = < parent component >-< component >-<ui element (button, input, dropdown)>-< state : optional (true/false) >